examples/android-room/src/main/java/io/sentry/android/roomsample/ui/LyricsActivity.kt

```diff
@@ -0,0 +1,60 @@
+package io.sentry.android.roomsample.ui
+
+import android.annotation.SuppressLint
+import android.net.Uri
+import android.os.Bundle
+import android.widget.EditText
+import androidx.activity.ComponentActivity
+import androidx.appcompat.widget.Toolbar
+import io.sentry.ITransaction
+import io.sentry.Sentry
+import io.sentry.SpanStatus
+import io.sentry.android.roomsample.R
+import io.sentry.android.roomsample.data.Track
+import java.io.File
+import java.io.FileInputStream
+
+@SuppressLint("SetTextI18n")
+class LyricsActivity : ComponentActivity() {
+    private lateinit var file: File
+    private lateinit var lyricsInput: EditText
+    private lateinit var transaction: ITransaction
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_lyrics)
+
+        transaction = Sentry.startTransaction(
+            "Track Interaction",
+            "ui.action.lyrics",
+            true
+        )
+
+        lyricsInput = findViewById(R.id.lyrics)
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
+
+        val track: Track = intent.getSerializableExtra(TRACK_EXTRA_KEY) as Track
+        toolbar.title = "Lyrics for ${track.name}"
+
+        val dir = File("$filesDir${File.separatorChar}lyrics")
+        dir.mkdirs()
+
+        file = File(dir, "${track.id}.txt")
+        if (file.exists()) {
+            lyricsInput.setText(file.readText())
+        }
+    }
+
+    override fun onBackPressed() {
+        if (!file.exists()) {
+            file.createNewFile()
+        }
+        file.writeText(lyricsInput.text.toString())
+        transaction.finish(SpanStatus.OK)
```

Comment:
timing of `transaction` wont make any sense here since `onBackPressed` can be clicked hours later or not even clicked, I'd rather use the auto transaction for `LyricsActivity`.
and a new transaction for `onBackPressed` if none set in the scope.